python/cugraph/cugraph/link_prediction/jaccard.py

```diff
@@ -11,16 +11,45 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import cudf
-from cugraph.link_prediction import jaccard_wrapper
 from cugraph.utilities import (
     ensure_cugraph_obj_for_nx,
     df_edge_score_to_dictionary,
     renumber_vertex_pair,
 )
+import cudf
+import warnings
+
+from pylibcugraph import (
+    jaccard_coefficients as pylibcugraph_jaccard_coefficients,
+)
+from pylibcugraph import ResourceHandle
+
+
+# FIXME: Move this function to the utility module so that it can be
+# shared by other algos
+def ensure_valid_dtype(input_graph, vertex_pair):
+    vertex_dtype = input_graph.edgelist.edgelist_df.dtypes[0]
+    vertex_pair_dtypes = vertex_pair.dtypes
+
+    if vertex_pair_dtypes[0] != vertex_dtype or vertex_pair_dtypes[1] != vertex_dtype:
+        warning_msg = (
+            "Jaccard requires 'vertex_pair' to match the graph's 'vertex' type. "
+            f"input graph's vertex type is: {vertex_dtype} and got "
+            f"'vertex_pair' of type: {vertex_pair_dtypes}."
+        )
+        warnings.warn(warning_msg, UserWarning)
+        vertex_pair = vertex_pair.astype(vertex_dtype)
+
+    return vertex_pair
 
 
-def jaccard(input_graph, vertex_pair=None, do_expensive_check=True):
+# FIXME:
+# 1. Be consistent with column names for output/result DataFrame
+# 2. Enforce that 'vertex_pair' is a cudf Dataframe with only columns for vertex pairs
+# 3. We need to add support for multi-column vertices
+
+
+def jaccard(input_graph, vertex_pair=None, do_expensive_check=False, use_weight=False):
```

Comment:
We have a requirement that all new or updated user APIs have type annotations.  [Here](https://github.com/rapidsai/cugraph/blob/98324acc83a55618f1fa41313b4bd2ec32079215/python/cugraph/cugraph/centrality/betweenness_centrality.py#L31-L40)'s an example.